classes/mail/Mailable.inc.php

```diff
@@ -135,6 +142,26 @@ public function build() : self
         return $this;
     }
 
+    /**
+     * @return string localized Mailable name
+     */
+    public static function getName(): string
+    {
+        return !empty(static::$name) ?
+            __(static::$name) :
+            '####';
```

Comment:
I'm thinking about when/if we might want to use a `Mailable` but not have a custom class `Example extends Mailable`. For example, if I wanted to send an email and I didn't need it to be customized by the journal, I might just use the `Mailable` class directly:

```php
$mailable = new Mailable();
$mailable
    ->setRecipients([])
    ->setSender($user)
    ->subject($subject)
    ->body($body);
Mail::send($mailable);
```

I imagine that this would be used, for example, when sending an email at the site level. There the admin can't customize email templates anyway, so such a mailable wouldn't need to use the whole mailable system.

Do you think it makes sense to split out something like a `Mailable` from a "configurable" `Mailable`? I was thinking we could do this with a trait:

```php
trait Configurable
{
    public static string $name = '';

    public static function getName(): string
    {
        return !empty(static::$name)
            ? __(static::$name)
            : throw new Exception('Configurable mailable created without a name.');
    }
}

class Example extends Mailable
{
    use Configurable;

    public static string $name = 'my.name';
}
```

Maybe it's an unnecessary abstraction. But what do you think?

Reply:
It sounds reasonable. 
As I found traits aren't designed to hold properties, which are expected to be declared by a class that uses them. I've slightly changed this approach. The Mailable class declares `name` and `description` protected properties, which have a null default value.